File: src-tauri/src/commands/import.rs
use crate::db::DbState;
use crate::service;
use tauri::State;

/// Integrated rate_card command: SM2 + update + review log in one invoke.
#[tauri::command]
pub fn rate_card(
    state: State<'_, DbState>,
    card_id: String,
    quality: i64,
) -> Result<service::study::RateResult, String> {
    let conn = state.conn.lock().map_err(|e| e.to_string())?;
    service::study::rate_card(&conn, &card_id, quality)
}

/// Parse text content (format detection + parsing + validation).
#[tauri::command]
pub fn parse_txt_content(text: String) -> Result<service::import::ParseResult, String> {
    Ok(service::import::parse_txt_content(&text))
}

/// Import from raw text: parse + create deck + import cards, all in one operation.
#[tauri::command]
pub fn import_from_text(
    state: State<'_, DbState>,
    deck_name: String,
    text: String,
) -> Result<service::import::ImportFromTextResult, String> {
    let conn = state.conn.lock().map_err(|e| e.to_string())?;
    service::import::import_from_text(&conn, &deck_name, &text)
}

File: src/api/import.ts
import { invoke } from '@tauri-apps/api/core'
import type { ParseResult } from '../types/generated/ParseResult'
import type { ImportFromTextResult } from '../types/generated/ImportFromTextResult'

export async function parseTxtContent(text: string): Promise<ParseResult> {
  return await invoke<ParseResult>('parse_txt_content', { text })
}

export async function importFromText(
  deckName: string,
  text: string,
): Promise<ImportFromTextResult> {
  return await invoke<ImportFromTextResult>('import_from_text', { deckName, text })
}
